src/diagnostic_variables.jl

```diff
@@ -66,9 +66,44 @@ function GridVariables(G::GeoSpectral{NF}) where NF
                         u_grid,v_grid,temp_grid_anomaly)
 end
 
+"""
+Struct holding quantities calculated from the physical parameterisations. All quantities
+are in grid-point space.
+"""
+struct ParametrizationVariables{NF<:AbstractFloat}
+    sat_vap_pressure   ::Array{NF,3}   # Saturation vapour pressure
+    sat_spec_humidity  ::Array{NF,3}   # Saturation specific humidity
+    cloud_top          ::Array{Int,2}  # Cloud-top
+    precip_large_scale ::Array{NF,2}   # Large-scale precipitation
+    humid_tend         ::Array{NF,3}   # Humidity tendencies due to physics
+    temp_tend          ::Array{NF,3}   # Temperature tendencies due to physics
```

Comment:
So the `humid_tend` and `temp_tend` which are already defined in `Tendencies` are in spectral space. 

My original preference for the parameterisations was to simply add up the tendencies as we went along, rather than allocating a new array for each tendency and parameterisation and summing them at the end. However the problem I ran into here (which may be repeated for other parameterisations) is that the humidity tendency due to LSC gets re-used to calculate precipitation. So you have to allocate it either.

For now, I think I would just pre-allocate a new array for each tendency and parameterisation (`humid_tend_lsc`, `temp_tend_lsc`, ...) and then at the end we can optimise it, once we have a clearer picture of how the parameterisations interact?

Reply:
>However the problem I ran into here (which may be repeated for other parameterisations) is that the humidity tendency due to LSC gets re-used to calculate precipitation. So you have to allocate it either.

I think this is exactly where we have to start thinking about working in column vectors rather than horizontal fields. That would solve many problems. I reckon it might actually be the best to start very early with this restructuring from vert x lon x lat to lon x lat x vert (outer to inner loop). Because then there's no problem to allocate a bunch of vectors for intermediate calculations that get reused for the next horizontal grid point. That saves memory and gives us more flexibility to reuse stuff.

Reply:
Yeah that's a great point, no problem allocating an 8-element vector.